README.md

```diff
@@ -178,7 +178,83 @@ Please note that SendStartedStatement and SendCompletedStatement are both firing
 
 In the OnApplicationQuit call, I showed an example of how to call the publisher with a custom verb. OnApplicationQuit is a hook that will fire when the user quits the application. If you go ahead and attempt to close out of the player you'll see it fire off stuff again.
 
-There's also a way to make fully custom statements but I'm still trying to figure out the best way to make it clear on how to construct the statements from scratch and what gets included in them.
+
+### Statement Hooks
+
+Sometimes in your unity application you need to have access to a given statement for one reason or another. For this reason there's a way to hook into the statement sendoff from within the application. The following example shows how this is done.
+
+```csharp
+using System;
+using UnityEngine;
+using LRS.Domain;
+using System.Text.Json.Nodes;
+
+class Hook : MonoBehaviour
+{
+
+    // function that fires every time a statement is sent off.
+    private void OnStatementSent(JsonObject statement)
+    {
+        Debug.Log(statement["verb"]["id"]);
+    }
+
+    // Unity boilerplate for adding hooks to handlers
+    private void OnEnable()
+    {
+        Publisher.OnStatementSent += OnStatementSent;
+    }
```

Comment:
Is this some sort of weird C# syntax? I don't know if `OnStatementSent` is supposed to be a boolean, some sort of integer, or an actual object instance.

Reply:
It's unity boilerplate for adding hooks to handlers. Most things in Unity are doing this under the hood (any time you call Update, Start, Awake, etc you're just adding a hook to a handler) but since this isn't a native Unity hook we need to declare the hook explicitly and also do the cleanup.

Reply:
Oh I was specifically asking about the `+=` and `-=` syntax.

Reply:
I guess if i wanted to be less verbose here I would declare a class (similar to Unity's `MonoBehavior`) that has the hooking in the class definition, but since most people are gonna want to be using this from within a class that inherits from `MonoBehavior` already (since scripting in Unity is almost always wrapped by `MonoBehavior`) it would probably make things way more obtuse and hard to deal with.

Reply:
+= and -= are both binary operators that take an argument and either append it or remove it from an aggregate or list.

Reply:
So `OnStatementSent` is a collection...and you're adding/removing it from itself?